### DISCUSSION TITLE
Microsoft Copilot vs Github Copilot

### DISCUSSION THREAD

**Discussion by JustinTBrown** [community/community#103027] [New to GitHub] [answered]
If I already pay for Copilot Pro through Microsoft, do I have to pay separately through Github too? That feels weird since it's branded as one product.

**Comment by Vijaykv5** [comment]


**Reply by ramazanpolat** [reply]
Nope! This is NOT the case for **Github Copilot** vs **Copilot Pro**. For example, Copilot Pro is more expensive ($20.00 user/month) while Github Copilot is $10.00 user/month. In exchange, Copilot Pro has Office 365 integration and that's all. However both products having Copilot as a name is not just confusing but also a stupid decision!

**Comment by JustinTBrown** [comment, marked as answer]
I already pay for both and called into Microsoft Support. They were confused by my question, and kept interpreting it as an issue with double billing. They also mentioned they don't have any visibility into Github billing. I think that suggests MS treats the two Copilot products separately.

**Reply by drmssst** [reply]
I asked Github support as well as researched what Copilot Pro covered, and the outcome is that Github Copilot is a separate subscription. There are no discounts or considerations taken into account should you have one and wish to subscribe to the other. Below is a copy & paste from the Github support agent response:

_Thank you for contacting GitHub Support. I'm here to assist you with using GitHub Copilot.

Upon reviewing your your account, it appears that the Copilot license is not enabled. It's possible that it was activated for a different account, or you may have purchased Copilot Pro from Microsoft.

Before you can utilize GitHub Copilot with your personal account, you'll need to initiate a free trial or subscribe to GitHub Copilot Individual. We have a useful resource to help you get started with Copilot: Quickstart for GitHub Copilot in the GitHub Docs.

However, if you are referring to Microsoft Copilot Pro, I recommend reaching out to the Microsoft support team for assistance with Copilot Pro: https://support.microsoft.com/en-us/copilot-pro_

**Reply by MFerguson1** [reply]
Can I withdraw money's from my account

**Comment by shahinkhorshidfar** [comment]
[I](url)

**Comment by ProFive** [comment]
VS code can not integrate with GitHub Copilot
Your project contains issues that can be fixed by running "dart fix" from the command line.

**Comment by nadila2k** [comment]
is this good

**Comment by kamyee2025** [comment]
Just for kicks, I asked Copilot itself, and this was the last part of the response :-)

In summary, if you pay for **Microsoft Copilot**, it does not automatically cover the cost of **GitHub Copilot**. They are separate services with distinct pricing models. Choose the one that best aligns with your needs and development environment! 🚀

Source: Conversation with Bing, 3/19/[2](https://www.techtarget.com/searchEnterpriseDesktop/tip/Comparing-Copilot-for-Microsoft-365-vs-GitHub-Copilot)024
(1) Microsoft [3](https://www.gyata.ai/github/github-copilot-cost/)65 Copilot vs GitHub Copilot. Comparison, Differences .... https://www.o365cloudexperts.com/blog/microsoft-365-copilot-vs-github-copilot/.
(2) Comparing Copilot for Microsoft 365 vs. GitHub Copilot. https://www.techtarget.com/searchEnterpriseDesktop/tip/Comparing-Copilot-for-Microsoft-365-vs-GitHub-Copilot.
(3) Understanding the Cost of GitHub Copilot: Free vs Paid. https://www.gyata.ai/github/github-copilot-cost/.
(4) . https://bing.com/search?q=Microsoft+Copilot+vs+GitHub+Copilot+pricing.
(5) undefined. https://www.zdnet.com/article/what-is-microsoft-copilot-heres-everything-you-need-to-know/.
(6) undefined. https://www.uctoday.com/collaboration/is-microsoft-copilot-free-the-complete-guide-to-copilot-pricing/.

**Comment by Antebios** [comment]
Well, there SHOULD either be bundled or given a discount for having both services.

**Comment by SkyLeach** [comment]
I just ran into this.  I have copilot 365 as well and I'm using it to write code, but when I try to use observable my free tier is crap... essentially nonfunctional.

Thing is, I want to use it for observable and I'm using observable both online and offline.  The online version MUST use github copilot while the offline MUST use microsoft copilot... and the sessions exist in their respective session stores/associations.

It's not just the price, it's also the confusion and mess that comes from having online and offline versions of copilot that don't play together and quite possibly use completely different models.

If I were any kind of corporate entity I'd be tempted to throw together ollama on AWS or Azure with an uncensored llama 2 or other 2048++ token model (mistral maybe).  The idea being: predictable version control and tool chain not subject to unreliable session storage and security.

In code/vim/neovim I'm not going to have an issue because I can pick my private LLM in those, but not github so... even bigger mess now.

**Comment by ForbiddenEra** [comment]
Microsoft, fix this shit. One should give you access to the other and vice versa. WTF

**Comment by sekarbala123** [comment]
Why is Microsoft have'nt even thought this through? This so confusing, terribly worse when we have to pay and find out which we cannot use? And specifically in vscode, lot many jorgans with copilot and azure/ms copilot mixing that i no/know more understand to draw a lion/line on this?! Thanks for your deep expertise MS!

**Reply by ForbiddenEra** [reply]
> Why is Microsoft have'nt even thought this through? This so confusing, terribly worse when we have to pay and find out which we cannot use? And specifically in vscode, lot many jorgans with copilot and azure/ms copilot mixing that i no/know more understand to draw a lion/line on this?! Thanks for your deep expertise MS!

right? they could at least have some differentiation in the fucking names of the product..

even more confusing for anyone that actually thinks copilot is a model, lmao
